script/hassfest/translations.py

```diff
@@ -47,7 +47,7 @@ def check_translations_directory_name(integration: Integration) -> None:
         return
 
     if legacy_translations.is_dir():
-        integration.add_warning("translations", MOVED_TRANSLATIONS_DIRECTORY_MSG)
+        integration.add_error("translations", MOVED_TRANSLATIONS_DIRECTORY_MSG)
```

Comment:
The warning is kept but changed to error, so repositories using the hassfest action now get a failed action run.